### PR TITLE
ESLint additional rules

### PR DESCRIPTION
**Pull request by jacogr**
Additional eslint rules so we can stop re-formatting code from others due to differences in interpretation. (Auto-fixes done on failed checks with `eslint --fix` - insubstantial changes, although a large number automatically found & fixed.)

1. JSX tag spacing, closing tag to align with opening (if multiple lines)

```js
<SomeTag
  attribute={ something }
  className={ styles.item }
/>
```

2. No unneeded block spacing

```js
class Something extends Component {
  static propTypes = { // no blank line before
    ...
  }

  render () { // no blank line padding function after
    if (x == y) { 
       doSomething(); // no blank line before
    }

    switch (foo) {
      case 'bar': // no blank line before
        return <div></div>;
    }
  }
}
```

### CODE REVIEW

**Comment by ngotchac** [comment]
@derhuerst I like this convention, but you might want to say how you feel about it before it's carved in stone

File: js/src/dapps/githubhint/Application/application.js [hunk 1]
```diff
@@ -128,11 +131,13 @@ export default class Application extends Component {
               <Button
                 disabled={ registerBusy }
                 invert={ registerType !== 'file' }
-                onClick={ this.onClickTypeNormal }>File Link</Button>
+                onClick={ this.onClickTypeNormal }
+              >File Link</Button>
```
Comment (derhuerst) [inline js/src/dapps/githubhint/Application/application.js]: I know the diff has been produced automatically, but this is still not consistent with the other places. Consider adapting it. – Same below.

File: js/src/dapps/tokenreg/Accounts/AccountSelector/account-selector.js [hunk 2]
```diff
@@ -39,7 +38,8 @@ class AccountSelectorItem extends Component {
 
     const icon = (<IdentityIcon
```
Comment (derhuerst) [inline js/src/dapps/tokenreg/Accounts/AccountSelector/account-selector.js]: Consider another newline here.

File: js/src/dapps/tokenreg/Accounts/AccountSelector/container.js [hunk 3]
```diff
@@ -25,7 +25,7 @@ class AccountSelectorContainer extends Component {
   render () {
     return (<AccountSelector
```
Comment (derhuerst) [inline js/src/dapps/tokenreg/Accounts/AccountSelector/container.js]: As above, consider another newline here.

File: js/src/dapps/tokenreg/Actions/Query/query.js [hunk 4]
```diff
@@ -155,7 +160,8 @@ export default class QueryAction extends Component {
 
             validationType={ SIMPLE_TOKEN_ADDRESS_TYPE }
             onChange={ this.onChange }
-            onEnter={ this.onQuery } />)
+            onEnter={ this.onQuery }
+             />)
           : (<InputText
```
Comment (derhuerst) [inline js/src/dapps/tokenreg/Actions/Query/query.js]: As above, consider another newline here.

File: js/src/dapps/tokenreg/Actions/container.js [hunk 5]
```diff
@@ -22,11 +22,10 @@ import Actions from './component';
 import { registerToken, registerReset, queryToken, queryReset } from './actions';
 
 class TokensContainer extends Component {
-
   render () {
     return (<Actions
```
Comment (derhuerst) [inline js/src/dapps/tokenreg/Actions/container.js]: As above, consider another newline here.

File: js/src/dapps/tokenreg/Container.js [hunk 6]
```diff
@@ -39,7 +39,7 @@ class Container extends Component {
     return (<Application
```
Comment (derhuerst) [inline js/src/dapps/tokenreg/Container.js]: As above, consider another newline here.

File: js/src/dapps/tokenreg/Inputs/Text/container.js [hunk 7]
```diff
@@ -20,11 +20,10 @@ import { connect } from 'react-redux';
 import InputText from './input-text';
 
 class InputTextContainer extends Component {
-
   render () {
     return (<InputText
```
Comment (derhuerst) [inline js/src/dapps/tokenreg/Inputs/Text/container.js]: As above, consider another newline here.

File: js/src/views/WriteContract/writeContract.js [hunk 8]
```diff
@@ -277,7 +276,7 @@ class WriteContract extends Component {
               label='Deploy'
```
Comment (derhuerst) [inline js/src/views/WriteContract/writeContract.js]: This indentation feels inconsistent with the rest.

**Comment by derhuerst** [comment]
@ngotchac This convention makes sense to me. What I've been having trouble with is *how*/*which* variables/props to group, not how much spacing etc.

**Comment by jacogr** [comment]
@derhuerst Thanks for the run-though, will look at the ones that jumped out, may as well help Mrs. Auto a bit for future sanity.

**Pull request merged**